api/src/main/java/org/pmiops/workbench/api/ProfileController.java

```diff
@@ -275,12 +275,9 @@ private User initializeUserIfNeeded() {
     user.setGivenName(request.getGivenName());
     userDao.save(user);
 
-    try {
-      return ResponseEntity.status(HttpStatus.CREATED).body(profileService.getProfile(user));
-    } catch (ApiException e) {
-      log.log(Level.SEVERE, "Error getting user profile: {0}".format(e.getResponseBody()), e);
-      return ResponseEntity.status(e.getCode()).build();
-    }
+    // TODO(dmohs): This should be 201 Created with no body, but the UI's swagger-generated code
```

Comment:
So it's pretty normal for create and update REST API calls to return the object that just got created / updated, so the client can cache state that it may not have provided in the original request without making another RPC. That's what I had it doing before. (I know profileService.getUser() wasn't working before for non-logged-in users... but we could certainly fix that.)

Is there a reason you'd rather not return it?

Reply:
Some discussion of this topic:

https://stackoverflow.com/questions/797834/should-a-restful-put-operation-return-something

Anyway, if the client doesn't need this data, I don't mind too much either way.

Reply:
One reason we might want to return something in the response: to let the client know what e-mail address was just created for the user. That way we can prompt them to sign in with that specific account, without having to know what domain the server is using.

Reply:
I couldn't think of anything the server knew that the client didn't just tell it, so returning something would seem like a duplication of data (so then the client has to decide what to do if it doesn't agree). Maybe the domain makes sense, though the UI will definitely know it so it can inform the user during account creation. I guess someone using the API directly wouldn't. Are we supporting that use case?

Reply:
I don't think this is an endpoint we're going to want people to call from other places particularly.

I just figured having the client learn that the domain in use is allofusresearch.org@ or fake-domain.org@ or whatever is best handled by having the server return something to it.

But if the UI needs to display the domain when the user is filling out the form to sign up, then I guess the client would need to know before this point anyway (maybe through some other endpoint?)... so it kinda depends on the UX.

There isn't any other information at this stage that the client would care about, so if we want to return nothing that's cool.

Reply:
Jeremy suggested (and I agree) that the username field should look like:
[input] @fake-research-aou.org
(or whatever domain)